### PR TITLE
Complete Poster Task pages with Router Navigation and data sharing between pages.

### LINKED ISSUE #17: Poster's tasks page
## Basic Information
Create the poster's (person making the task) tasks page   
Poster can create tasks and check bid here
[Figma design](https://www.figma.com/design/bhSe5HkMJKKN0Bs1ZqSddi/Penni?node-id=5996-22124&t=w0t63mb6QaHbIOjN-0)

(Frame 1549)

### PR DESCRIPTION
**Pull request by ArcueidShiki**
## Change Summary
**Complete Poster Task pages with Router Navigation and data sharing between pages.**
1. This PR includes UI of poster tasks pages, Router Navigation, Data sharing between pages.
2. TODO placeholders are for api data fetching part.
### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [#17 ] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [x] The change has documentation

## Other Information
image
image
image
image
image
image

# Related issue

- Resolve #17

### CODE REVIEW

**Comment by ArcueidShiki** [comment]
Payment page is same with issue #122 another team is working on right now.

**Comment by yunho7687** [comment]
@Julie-Salazar yo, could you please upload a screenshot or just tell us why the Vercel deployment has failed
Screenshot 2024-07-17 at 9 13 33 PM

**Comment by ArcueidShiki** [comment]
> @Julie-Salazar yo, could you please upload a screenshot or just tell us why the Vercel deployment has failed Screenshot 2024-07-17 at 9 13 33 PM

image
without permit

**Comment by Julie-Salazar** [comment]
> > @Julie-Salazar yo, could you please upload a screenshot or just tell us why the Vercel deployment has failed Screenshot 2024-07-17 at 9 13 33 PM
> 
> image without permit

you cant do it with your student email you have to use cfc email to enter pls dm dylan for creds @dct0 

> > @Julie-Salazar yo, could you please upload a screenshot or just tell us why the Vercel deployment has failed Screenshot 2024-07-17 at 9 13 33 PM
> 
> image without permit

you can't log in with your student email you need to use cfc email >> ask dylan on discord if you need creds

File: client/src/pages/poster/tasks/[taskid]/[bidderid]/bid-details.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { Button } from "@/components/ui/button";
+import Header from "@/components/ui/header";
+import PersonDetail from "@/components/ui/person-detail";
+
+export default function BidDetail() {
+  const router = useRouter();
+  if (router.query.bidder === undefined || router.query.taskid === undefined) {
+    return <>Loading...</>;
+  }
+  const bidder = JSON.parse(String(router.query.bidder));
```
Comment (yunho7687) [inline client/src/pages/poster/tasks/[taskid]/[bidderid]/bid-details.tsx:13]: could you please explain what `router.query.bidder` is or do you mean `router.query.bidderid` instead?

Comment (ArcueidShiki) [reply client/src/pages/poster/tasks/[taskid]/[bidderid]/bid-details.tsx:13]: The bidder json string was obtained from last page task-details page. I won't do a get request from backend, by data you've alreay got from previous pages, sharing between different pages.
For exmaple, bio, name, profile data of bidder just obtain from last pages, there's no need to call api of get request.

File: client/src/pages/poster/tasks/task-list.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+type TaskListProp = {
+  className?: string;
+};
+
+export default function TaskList({ className }: TaskListProp) {
+  const router = useRouter();
+  // TODO api fetch tasklist replace dummy data.in reverse order.
+  // pass taskid and title to next page.
+  function NavigateWithData(task: Task): void {
```
Comment (yunho7687) [inline client/src/pages/poster/tasks/task-list.tsx]: according to the convention, the better practice is to use a arrow function here. please refer to the style-guide.md in client fold.

File: client/src/pages/poster/tasks/[taskid]/[bidderid]/bid-details.tsx (hunk 1 above)
Comment (yunho7687) [reply client/src/pages/poster/tasks/[taskid]/[bidderid]/bid-details.tsx:13]: cool I got it! well done

File: client/src/pages/poster/tasks/task-list.tsx (hunk 2 above)
Comment (ArcueidShiki) [reply client/src/pages/poster/tasks/task-list.tsx]: Modified

**Review by yunho7687 — APPROVED** [review]

**Pull request merged**